### ISSUE TITLE
Issue setting graphic on JFXButton

### ISSUE THREAD

**Issue by matthew-matvei** [jfoenixadmin/JFoenix#221]
Hi again, same project different problem :p 

I am able to set the graphic of a regular `javafx.scene.control.Button` (that has a set background) to a Text object (on which I can use a drop shadow effect) without any issues. However, when I do the same with a JFXButton, the appearance is fine, but a click event handler does not work. Some redacted code:

``` java
public static Button buildMusicButton(File musicDirectory) {
    Text text = new Text(musicDirectory.toString());
    Button button = new Button();
    button.setGraphic(text);
    button.setBackground(getBackgroundImage());

    return button;
}
```

This gives me a fully-functional button (the click event handler defined outside this method). Whereas the same, using:

``` java
public static JFXButton buildMusicButton(File musicDirectory) {
    Text text = new Text(musicDirectory.toString());
    JFXButton button = new JFXButton();
    button.setGraphic(text);
    button.setButtonType(ButtonType.RAISED);
    button.setBackground(getBackgroundImage());

    return button;
}
```

produces a button that, even though the type has been set to raised, shows no drop shadow and with no click functionality. Strangely enough, if I navigate using different buttons (those that at this stage do not have a background image set), then come back to this same button, it does work perfectly. It appears to have the issue only during the program's initialisation.

So I think there may be some conflict with the JFXButton or the JFXButtonSkin and setting a graphic on a button that has a set background, at least when the program is being initialised. Perhaps it prevents the click event handler being attached properly, or maybe the graphic simply 'intercepts' the click in some way. I've provided a screenshot of the application to give some visual context:

![screenshot from 2017-01-10 09 28 38](https://cloud.githubusercontent.com/assets/19278663/21787854/38e12274-d717-11e6-9391-389de0cf36d9.png)

*Note: that displays my current 'solution', which is just to go ahead with a regular Button for now*

Unfortunately, there is no stack trace, and I'm unsure of what to be looking for whilst debugging.

Let me know if you require any other info.

**Comment by jfoenixadmin** [comment]
Hello,
We were not able to reproduce this issue using JFoenix latest version.  can please test again using the latest version.
if the issue still persist plz provide more details about your code and the body of getBackgroundImage() in order to detect what might be causing the issue.

Regards,

**Comment by matthew-matvei** [comment]
Thanks for the quick reply.

I've updated by downloading the `.jar` file available in the README and still have the issue. Is this the most recent stable version?

I'll provide the code for the `getBackgroundImage()` method, and then I'll work on making a small example in a separate repository to give a more concise example. In the mean time, if you notice anything in this code, let me know:

``` java
public static Background getBackgroundImage(URI imagePath) {

    BackgroundImage image = new BackgroundImage(
        new Image(imagePath.toString(), true, BackgroundRepeat.NO_REPEAT,
        BackgroundRepeat.NO_REPEAT, BackgroundPosition.CENTER,
        new BackgroundSize(100, 100, true, true, false, true));

    Background background = new Background(image);

    return background;
}
```

**Comment by matthew-matvei** [comment]
Unfortunately, I'm unable to reproduce the issue using a simplified copy. The JFXButton's behaviour is definitely unable to do what the regular Button is able to, evidenced when I swap in a regular button, but I can't see what might be causing that, and it's definitely a very specific problem.

If you've got no comment on my previous question, you can close this issue.

**Comment by jfoenixadmin** [comment]
well if provide a sample code of your application that shows the issue, we might be able to help.

**Comment by matthew-matvei** [comment]
It seems rather complicated. I recreated a simple version of the application, but was unable to reproduce the issue. If you're interested, my project in full can be found [here](https://github.com/matthew-matvei/All-Music-No-Labels). Changing all Button objects to JFXButton ones in `MusicGuiBuilder#buildMusicButton()` will show the issue.

The problem must be a strange combination of elements, though tracing the code responsible for those buttons didn't yield any insight.

**Comment by jfoenixadmin** [comment]
Hello,
I checked your application, and changed the method buildMusicButton() to use JFXButtons, it seems working fine however for some reason there is a drop shadow effect that is being added due to your css or styling. also I noticed that the 1st button when i opened the browser tab is java Button not JFXButton.

so the problem most probably related to your ui architecture and css, as we were not able to reproduce using a simple example.

Regards,
